### PR TITLE
[basic.align] Move the Alignment clause adjacent to the Object model

### PR DESCRIPTION
**Pull request by AlisdairM**
[Edit]
Simplified to move just the Alignment subclause.

[Original, for reference]
Reorder the subclauses of [basic.memobj], Memory and objects, into a logical progression where each clause augments the one before.

* Memory model specifies the notion of memory for C++
* Object model specifies objects, and how they occupy memory
* Alignment restricts where objects can reside in memory
* Storage duration describes how objects acquire well-aligned memory
* Intermediate and erroneous values specifies the initial contents of acquired memory
* Lifetime describes the validity of objects within their acquired storage
* Temporary objects have a lifetime of their own.

### CODE REVIEW

**Comment by jensmaurer** [comment]
I don't think micro-optimizing various small-ish subclauses in the standard is a good way to spend our time.

That said, I disagree with the suggested ordering.  In particular, storage duration is more about when the destructor is automatically called in various circumstances (or not at all), and not so much about the actual duration of the storage. I think we first should know about lifetime in general before we dig into that.

Further, indeterminate / erroneous values is mostly discussing what happens within the lifetime of an object when no initialization has happened, and not about storage without in-lifetime objects.

In short, I don't find the current ordering "obviously wrong".

**Comment by AlisdairM** [comment]
How would you feel about moving just the alignment clause to follow Object model?

**Comment by jensmaurer** [comment]
> How would you feel about moving just the alignment clause to follow Object model?

I'm ok with that.  Alignment is an aspect of the object model, I think.

**Review by jensmaurer — COMMENTED** [review]
Please rebase; there's a stray commit here.

**Comment by AlisdairM** [comment]
I believe the stay commit _is_ a consequence of rebasing?

**Comment by jensmaurer** [comment]
I hope that isn't true.  I just merged the change that is your second commit to "main", so I'm a bit surprised to see it pop up here (again).

**Comment by AlisdairM** [comment]
I just rebased, and the commit ID of that intermediate commit has changed.  I rebased again and that should have fixed it.  I have no idea what the commit ID would change, but I guess that is what I get for jumping on a change that just landed?

**Comment by tkoeppe** [comment]
Please simplify the commit description, too (or at least during squashing).

**Comment by AlisdairM** [comment]
The commit description has been greatly simplified.  I retained the original description above _only_ in the GitHub conversation, so that Jens's comments retained their context.

**Comment by AlisdairM** [comment]
Oh, unless you mean the one-line description, I can fix that too.

**Comment by AlisdairM** [comment]
I hope that the PR is now in a simple and clean state, ready for landing.

**Comment by AlisdairM** [comment]
This should be properly rebased again.

**Comment by tkoeppe** [comment]
Applied via 2981bd94f25ea2199fd6b8af7aa76e03cf427697, thank you!
